### DISCUSSION TITLE
Composition HUB (C++)

### DISCUSSION THREAD

**Discussion by HamedSabri-adsk** [Autodesk/maya-usd#1738] [Ideas]
Hi guys,

I have been working with some USD assets recently and from time to time I need to see the full composition hierarchy and inspect data inside Maya. The old python comp that I wrote ages ago didn't cut it and also lacked support for sublayers.

I recently wrote a new composition inspector in C++ and thought to share it with you guys. The new version supports all comp arcs and represent composition via tree hierarchy ( similar to Solaris and UsdView ).

I recall that there was already a plan for composition hub and you may have a different design in mind. Nonetheless, I think the changes that I have would be a good start to build on top of. 

**Demo**

![composition_hub_demo](https://user-images.githubusercontent.com/48299423/135622622-bee6e684-b179-4409-9505-65ff953d3d62.gif)
:

**Comment by kxl-adsk** [comment]
Thank you for sharing, Hamed. Great to see you here!
